.github/workflows/artifacts-ci-lambda.yml

```diff
@@ -0,0 +1,116 @@
+      s3_bucket:
+        description: "Bucket name where lambda code is uploaded"
+        type: string
+        default: 'orbit-artifacts'
+      s3_bucket_key:
+        description: "Bucket name where lambda code is uploaded"
+        type: string
+        default: 'lambda'
+        required: false
+      platform:
+        description: "Change the platform on which runner will be working on "
+        required: false
+        type: string
+        default: ubuntu-latest
+      is_finished:
+        description: "Wait for called event to finish"
+        required: false
+        type: boolean
+        default: true
+    secrets:
+      gh_token:
+        required: true
+
+permissions:
+  id-token: none
+  issues: write
+  pull-requests: write
+  contents: read
+
+jobs:
+  build_push:
+    name: Run Repository Dispatcher
+    runs-on: ${{ inputs.platform }}
+    if: (github.event_name == 'push') || (github.event_name == 'pull_request' && contains(github.head_ref, 'release/')) || (github.event_name == 'workflow_dispatch')
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+          token:  ${{ secrets.gh_token }}
+      - name: Get repo data
+        id: repo_data
+        uses: htec-infra/gha-utils/actions/repo-data@v0.3.1
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ inputs.dotnet_version }}
+      - name: Compile code
+        run: |
+          dotnet publish ${{ inputs.solution_file }} -c Release -o ./dist
+      - name: Create zip archive
+        run: |
+          zip -qq -r -j ${{ steps.repo_data.outputs.app_version }}.zip ./dist/*
+      - name: upload lambda function
+        run: |
+            aws s3 cp ${{ steps.repo_data.outputs.app_version }}.zip s3://${{ inputs.s3_bucket }}/${{ inputs.s3_bucket_key }}/${{ inputs.app_name }}/
+      - name: Comment PR
+        id: comment_pr
+        if: github.event_name == 'pull_request'
+        uses: htec-infra/gha-utils/actions/comment-pr@v0.3.1
+        with:
+          github_token: "${{ secrets.GITHUB_TOKEN }}"
+    outputs:
+      app_version: ${{ steps.repo_data.outputs.app_version }}
+
+  auto_deploy:
+    name: Auto Deploy (Dev Only)
+    needs: [build_push]
+    runs-on: [ubuntu-latest]
+    if: contains(needs.build_push.outputs.app_version, 'alpha')
+    steps:
+      - name: Trigger deploy
+        uses: htec-infra/gha-utils/actions/repo-dispatch@v0.3.1
+        with:
+          gh_token: ${{ secrets.gh_token }}
+          repository: ${{ inputs.infra_repo }}
+          event_type: ${{ inputs.event_type }}
+          client_payload: '{"env": "${{ inputs.deploy_environment }}", "name": "${{ inputs.app_name || steps.repo_data.outputs.repo_name }}", "s3_zip_file": "${{ needs.build_push.outputs.app_version }}.zip", "s3_bucket": "${{ inputs.s3_bucket }}", "dry_run": "${{ inputs.dry_run }}"}'
```

Comment:
Do you need all this parameters?

Reply:
Unfortunately yes, they are all necessary in order to update lambda function. Maybe we could move s3_bucket parameter to other place but then we will lose some of the flexibility